test/python/pulse/compiler_passes/test_set_sequence.py

```diff
@@ -0,0 +1,379 @@
+    #     self.assertTrue((3, 1) in edge_list)
+    #
+
+    @named_data(*ddt_named_data)
+    @unpack
+    def test_parallel_instructions(self, alignment):
+        """test with two parallel instructions"""
+
+        ir_example = SequenceIR(alignment)
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(1), QubitFrame(1))))
+
+        ir_example = self._get_pm().run(ir_example)
+        edge_list = ir_example.sequence.edge_list()
+        self.assertEqual(len(edge_list), 4)
+        self.assertTrue((0, 2) in edge_list)
+        self.assertTrue((2, 1) in edge_list)
+        self.assertTrue((0, 3) in edge_list)
+        self.assertTrue((3, 1) in edge_list)
+
+    @named_data(*ddt_named_data)
+    @unpack
+    def test_sequential_instructions(self, alignment):
+        """test with two sequential instructions"""
+
+        ir_example = SequenceIR(alignment)
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+
+        ir_example = self._get_pm().run(ir_example)
+        edge_list = ir_example.sequence.edge_list()
+        self.assertEqual(len(edge_list), 3)
+        self.assertTrue((0, 2) in edge_list)
+        self.assertTrue((2, 3) in edge_list)
+        self.assertTrue((3, 1) in edge_list)
+
+    @named_data(*ddt_named_data)
+    @unpack
+    def test_pulse_target_instruction_broadcasting_to_children(self, alignment):
+        """test with an instruction which is defined on a PulseTarget and is
+        broadcasted to several children"""
+
+        ir_example = SequenceIR(alignment)
+        ir_example.append(Delay(100, target=Qubit(0)))
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(2))))
+
+        ir_example = self._get_pm().run(ir_example)
+        edge_list = ir_example.sequence.edge_list()
+        self.assertEqual(len(edge_list), 5)
+        self.assertTrue((0, 2) in edge_list)
+        self.assertTrue((2, 3) in edge_list)
+        self.assertTrue((2, 4) in edge_list)
+        self.assertTrue((3, 1) in edge_list)
+        self.assertTrue((4, 1) in edge_list)
+
+    @named_data(*ddt_named_data)
+    @unpack
+    def test_frame_instruction_broadcasting_to_children(self, alignment):
+        """test with an instruction which is defined on a Frame and is broadcasted to several children"""
+
+        ir_example = SequenceIR(alignment)
+        ir_example.append(ShiftPhase(100, frame=QubitFrame(0)))
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(0))))
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(1), QubitFrame(0))))
```

Comment:
Can you add one more instruction that doesn't take QubitFrame(0)? e.g. `MixedFrame(Qubit(1), QubitFrame(1))`